peer-signer/src/verifier.rs

```diff
@@ -0,0 +1,129 @@
+use clap::Parser;
+use libp2p::{identity::PublicKey, PeerId};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{str::FromStr, u64};
+
+#[derive(Debug, Parser)]
+#[clap(version = "1.0")]
+pub struct Config {
+    // Node to ask for network state
+    #[clap(long, default_value = "http://127.0.0.1:9933")]
+    pub node: String,
+
+    /// Peer Id of the peer we want to verify.
+    #[clap(long)]
+    pub peer_id: String,
+
+    /// Message that the signature of we want to check.
+    #[clap(long)]
+    pub message: String,
+
+    /// Hex encoded public key of the peer.
+    #[clap(long)]
+    pub public_key: String,
+
+    /// Signature we want to check.
+    #[clap(long)]
+    pub signature: String,
+
+    /// Max block difference with head.
+    #[clap(long, default_value = "10")]
+    pub block_difference: u64,
+}
+
+async fn make_request_and_parse_result<T: serde::de::DeserializeOwned>(
+    client: &Client,
+    endpoint: &str,
+    body: Value,
+) -> T {
+    let mut response: Value = client
+        .post(endpoint)
+        .json(&body)
+        .send()
+        .await
+        .expect("Storage request has failed")
+        .json()
+        .await
+        .expect("Could not deserialize response as JSON");
+    let result = response["result"].take();
+    serde_json::from_value(result).expect("Incompatible type of the result")
+}
+
+fn construct_json_body(method_name: &str, params: Value) -> Value {
+    serde_json::json!({
+        "jsonrpc": "2.0",
+        "id": 1,
+        "method": method_name,
+        "params": params
+    })
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct Peer {
+    best_hash: String,
+    best_number: u64,
+    peer_id: String,
+    roles: String,
```

Comment:
since you are already ignoring `protocol_version`, can we skip `roles` as well?

Reply:
Well, `protocol_version` is not skipped, as I take the result but indeed you can skip roles. I did that, and also skipped best_hash.